docs/s3.md

```diff
@@ -0,0 +1,153 @@
+# S3
+
+Action uses Amazon S3 for three purposes:
+
+   1. Serving manually managed assets, such as logos
+   2. Serving builds bundled with webpack
+   3. Storing and serving user-generated assets
+
+Action uses a single S3 bucket. Within the bucket, the directory structure
+is as follows:
+
+```
+   S3 Bucket
+   =========
+   |
+   |--- static/                     1) Manually managed assets
+   |
+   |--- :instance/
+      |--- build/                   2) Builds bundled with webpack
+      |  |--- :vX.Y.Z/
+      |
+      |--- store/                   3) User-generated assets
```

Comment:
OK, figured out what I don't like about it: we can't guarantee the folder structure unless we only give a client PUT access to _their_ folder. Otherwise, me, attacker matt, could put something into your folder. I could even do a distributed attack where I put 64^7 things in your folder to guarantee that your image gets replaced by my image. If i have write access to your folder, then what good is the folder hierarchy, anyways?